### DISCUSSION TITLE
Using subcommands and Arguments together

### DISCUSSION THREAD

**Discussion by Braayy** [Minestom/Minestom#1061] [Help] [answered]
How can i do something like this:
`/permission group <group name> add <permission name>`
`/permission group <group name> remove <permission name>`

I did a command called "permission" and a subcommand called "group" and it works fine, but now i need this "group name" argument, a "add" command and another argument "permission name".
Idk how to proceed with this

**Comment by GoldenStack** [comment, marked as answer]
You could try `ArgumentType.String("group")` and `ArgumentType.String("permission")` and give them custom suggestions via `Argument#setSuggestionCallback`
Here's an example of what I would do:

```java
Command group = ...;
ArgumentString groupName = ArgumentType.String("group");
groupName.setSuggestionCallback(...);

ArgumentString permission = ArgumentType.String("permission");
permission.setSuggestionCallback(...);

group.addSyntax(..., groupName, ArgumentType.Literal("add"), permission);
```
Similar code could be used for `remove`, but you'd obviously have to change the suggestion callback to actually suggest permissions it has.
